QModManager/API/SMLHelper/Assets/PDAItem.cs

```diff
@@ -0,0 +1,77 @@
+        /// </summary>
+        public abstract TechGroup GroupForPDA { get; }
+
+        /// <summary>
+        /// Override with the category within the group in the PDA blueprints where this item appears.
+        /// </summary>
+        public abstract TechCategory CategoryForPDA { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether <see cref="RequiredForUnlock"/> has been set to lock this blueprint behind another <see cref="TechType"/>.
+        /// </summary>
+        /// <value>
+        ///   Returns <c>true</c> if will be unlocked from the start of the game; otherwise, <c>false</c>.
+        /// </value>
+        /// <seealso cref="RequiredForUnlock"/>
+        public bool UnlockedAtStart => this.RequiredForUnlock == TechType.None;
+
+        /// <summary>
+        /// Message which should be shown when the item is unlocked. <para/>
+        /// If not overridden, the message will default to "<see langword="{FriendlyName} blueprint discovered!"/>"
+        /// </summary>
+        public virtual string DiscoverMessage => $"{this.FriendlyName} blueprint discovered!";
+
+        internal string DiscoverMessageKey => $"{TechType.AsString()}_DiscoverMessage";
+
+        /// <summary>
+        /// Initializes a new <see cref="PDAItem"/>, the basic class for any item that appears among your PDA blueprints. <para/>
+        /// DO NOT USE THIS CLASS DIRECTLY! Use <seealso cref="Craftable"/> or <see cref="Buildable"/> instead.
+        /// </summary>
+        /// <param name="classId">The main internal identifier for this item. Your item's <see cref="TechType" /> will be created using this name.</param>
+        /// <param name="friendlyName">The name displayed in-game for this item whether in the open world or in the inventory.</param>
+        /// <param name="description">The description for this item; Typically seen in the PDA, inventory, or crafting screens.</param>
+        internal PDAItem(string classId, string friendlyName, string description)
+            : base(classId, friendlyName, description)
+        {
+            CorePatchEvents += PatchTechDataEntry;
+
+            LanguageHandler.SetLanguageLine(DiscoverMessageKey, DiscoverMessage);
+        }
+
+        /// <summary>
+        /// This provides the <see cref="TechData"/> instance used to designate how this item is crafted or constructed.
+        /// </summary>
+        protected abstract TechData GetBlueprintRecipe();
+
+        private void PatchTechDataEntry()
+        {
+            CraftDataHandler.SetTechData(this.TechType, GetBlueprintRecipe());
+
+            CraftDataHandler.AddToGroup(this.GroupForPDA, this.CategoryForPDA, this.TechType);
+
+            if (this.UnlockedAtStart)
+                KnownTechHandler.UnlockOnStart(this.TechType);
+            else
+                KnownTechHandler.SetAnalysisTechEntry(this.RequiredForUnlock, new TechType[1] { this.TechType }, this.DiscoverMessageKey);
```

Comment:
I see you've changed this method call to pass in the Language strings key instead of the message.  
Did you first confirm that this would work?

Seeing as the commit that made the changes (  0913cb9f34f19f18b80fdfd44730f2964b10ac32) appears as _Verified_, I can't be sure.
Can you confirm that the `unlockMessage` field from `KnownTech.AnalysisTech` is pass through `Language.main.Get` instead of being used directly?

Reply:
0913cb9 is not verified, plus it includes built assemblies.
I am currently unable and unwilling to test since I have not finished the SMLHelper integration, but according to the game's code and KnownTechHandler, it will work.
Some of the methods in KnownTechHandler have a default parameter NotificationBlueprintUnlocked, which is a language key.

Reply:
> Some of the methods in KnownTechHandler have a default parameter NotificationBlueprintUnlocked, which is a language key. 

Works for me 👍